lib/searchablelist.fz

```diff
@@ -72,21 +72,31 @@ is
         c1 Cons => ((searchablelist c1.tail).find l) >>= n -> n + 1
 
 
-  # replace all occurances of f by r
+  # replace all occurrences of f by r
   #
-  replace (f, r list A) =>
-
-    # tail recursive helper
-    #
-    repl (h,       # the head part of from with f already replaced by r
-          t list A # the tail that still needs to be searched for f
-         ) list A
-    is
-      match (searchablelist t).find f
-        nil   => h ++ t
-        n i32 => repl h++(t.take n)++r (t.drop n+f.count)
-
-    repl nil from
+  replace (f, r list A) => replace f r nil from nil
+
+
+  # replace the first n occurrences of f by r
+  #
+  replace (f, r list A, n u64) => replace f r nil from n
+
+
+  # tail recursive helper for the replace features
+  #
+  private replace (f, r,
```

Comment:
I took a while for me to guess that f is find and r replace. A comment might help. I would actually prefer full words even but I think @fridis disagrees here.

Reply:
Hmm, right. We should not use verbs here. What about treasure and replacement?

Reply:
Old and new seems easy enough.